3rd/TokenBucket/TokenBucket.h

```diff
@@ -1,87 +1,39 @@
-/*
-Copyright (c) 2016 Erik Rigtorp <erik@rigtorp.se>
-
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to deal
-in the Software without restriction, including without limitation the rights
-to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in all
-copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
- */
 
 #pragma once
 
-#include <atomic>
 #include <chrono>
+#include <uvw/loop.hpp>
 
 class TokenBucket {
 public:
-    TokenBucket() : time_(0), timePerToken_(0), timePerBurst_(0) {}
-
-    TokenBucket(const uint64_t rate, const uint64_t burstSize) {
-        time_ = 0;
-        timePerToken_ = 1000000 / rate;
-        timePerBurst_ = burstSize * timePerToken_;
-    }
-
-    TokenBucket(const TokenBucket &other) {
-        time_ = other.time_.load();
-        timePerToken_ = other.timePerToken_.load();
-        timePerBurst_ = other.timePerBurst_.load();
-    }
+    TokenBucket() : rate_qps_(0), token_wallet_(0), lastFill_ms_(0) {}
```

Comment:
I think you can move this out of `3rd` directory as this has been essentially rewritten.

The coding style is a little off. The variable names should be snake case. And private variables should be prefixed with underscore (not suffixed).